crates/next-api/src/project.rs

```diff
@@ -183,8 +183,10 @@ impl ProjectContainer {
     #[turbo_tasks::function]
     pub fn new(options: ProjectOptions) -> Vc<Self> {
         ProjectContainer {
+            // we only need to enable versioning in dev mode, since build
+            // is assumed to be operating over a static snapshot
+            versioned_content_map: VersionedContentMap::new(options.dev),
```

Comment:
This looks a bit to complex to me. This could just be a `versioned_content_map: Option<Vc<VersionedContentMap>>`

```suggestion
            versioned_content_map: options.dev.then(|| VersionedContentMap::new()),
```

`get_versioned_content`, `get_source_map`, `hmr_content`, `hmr_identifiers` would just bail where there is no VersionedContentMap.
`emit_all_output_assets` would just call `emit_assets` when there is no VersionedContentMap.

Reply:
Does build not access VersionedContentMap at all?